src/uu/rm/src/rm.rs

```diff
@@ -250,14 +248,38 @@ fn remove(files: &[String], options: &Options) -> bool {
 
     for filename in files {
         let file = Path::new(filename);
+        // symlink_metadata's is_dir and is_file always returns false so we also need to grab a normal metadata
         had_err = match file.symlink_metadata() {
-            Ok(metadata) => {
-                if metadata.is_dir() {
-                    handle_dir(file, options)
-                } else if is_symlink_dir(&metadata) {
+            Ok(symlink_metadata) => {
+                if is_symlink_dir(&symlink_metadata) {
                     remove_dir(file, options)
-                } else {
+                } else if symlink_metadata.is_symlink() {
                     remove_file(file, options)
+                } else {
+                    match file.metadata() {
+                        Ok(metadata) => {
+                            if metadata.is_dir() {
+                                handle_dir(file, options)
+                            } else {
+                                remove_file(file, options)
+                            }
+                        }
+                        Err(_e) => {
```

Comment:
please add a test to cover this case

Reply:
There is already a test for it: test_rm::test_rm_symlink_dir() and test_rm::test_rm_invalid_symlink()

Reply:
the code coverage results don't agree ;)